src/Festung/Vault/VaultHandler.hs

```diff
@@ -109,6 +113,26 @@ query o v q = flattenError <$> sendMappedCommand f v q
 
 
 -- | Execute a parametrized query
-parametrizedQuery :: VaultOpener -> Vault -> String -> [P.Value] -> IO (Either VaultError P.QueryResult)
-parametrizedQuery o v q p = flattenError <$> sendMappedCommand f v (q, p)
+parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams
+                  -> IO (Either VaultError P.QueryResult)
+parametrizedQuery o v q (P.Params p) = flattenError <$> sendMappedCommand f v (q, p)
     where f = (,) o . ParametrizedQuery
+parametrizedQuery o v q (P.BulkParams bps) = 
+    _parametrizedQuery o v q (P.BulkParams bps) (Right P.emptyQueryResult)
+
+
+_parametrizedQuery :: VaultOpener -> Vault -> String -> P.QueryParams
+                   -> Either VaultError P.QueryResult -> IO (Either VaultError P.QueryResult)
+_parametrizedQuery o v q (P.BulkParams []      ) res    = return res
+_parametrizedQuery o v q (P.BulkParams (p:bp)) prevRes  = do
+    res <- parametrizedQuery o v q (P.Params p)
+    case res of
+        Left  _  -> return res
+        Right qr -> _parametrizedQuery o v q (P.BulkParams bp) (accumQueryResult prevRes res)
```

Comment:
I actually don't like the case expression here. I would write the function like this
```haskell
type Query = String
type QueryResult = Either VaultError P.QueryResult

_parametrizedQuery :: VaultOpener -> Vault -> Query -> P.QueryParams -> QueryResult -> IO (QueryResult)
_parametrizedQuery o v q _ (Left err) = return $ Left err
_parametrizedQuery o v q (P.BulkParams []) res = return res
_parametrizedQuery o v q (P.BulkParams (p:params)) prev = do
    res <- parametrizedQuery o v q (P.Params p)
    _parametrizedQuery o v q (P.BulkParams params) $ acc prev res
        where acc (Right p) (Right c)  = Right $ c { P.rowsChanged = (+ fromMaybe 0 (P.rowsChanged p)) <$> P.rowsChanged c }
              acc _         (Left err) = Left err
```